src/app/loadout/LoadoutPopup.tsx

```diff
@@ -255,6 +255,17 @@ class LoadoutPopup extends React.Component<Props> {
     );
   }
 
+  private renderLoadoutName(loadout) {
```

Comment:
I'm not generally a huge fan of breaking things out into a bunch of `renderX` helper functions - the further things are spread out and extracted into variables and functions the harder it is to track down what's actually happening (within reason, of course). I probably would have left this one inline.

Reply:
Done. I much prefer sections split out into their own render function with little to no logic in JSX, as long as things are obvious enough through naming. Each to their own though.

Reply:
Yeah, it’s not a bad way to go, but it does make it a bit harder to understand what’s happening all in one place. Once things get too complicated I tend to extract whole components rather than render methods.

Reply:
Yeah I'm with you on extracting components. I was refraining from refactoring too much until I understood your conventions.